### PR TITLE
fix(message-input): tooltip should be displayed when character limit has been reached

### PR DESCRIPTION
**Pull request by sanjeevdialpad**
fix(message-input): Tooltip should be displayed when character limit has been reached

## :hammer_and_wrench: Type Of Change

- [x] Fix
- [ ] Feature
- [ ] Refactoring
- [ ] Documentation

## :book: Description

Tooltip showing the character limit message from product should be displayed on the limit count when its negative.

### CODE REVIEW

**Comment by iropolo** [comment]
Hmm I cannot see the tooltip, how I can trigger it? seems like the params below are correct

image

**Comment by braddialpad** [comment]
> Hmm I cannot see the tooltip, how I can trigger it? seems like the params below are correct

It's the red -375 in the bottom right.

**Comment by braddialpad** [comment]
> It's the red -375 in the bottom right.

Nevermind you meant the tooltip not the number.. Yeah I also can't see it.

**Comment by sanjeevdialpad** [comment]
> > Hmm I cannot see the tooltip, how I can trigger it? seems like the params below are correct
> 
> 
> 
> It's the red -375 in the bottom right.
> 
> 
> 
> 

Yes is it not triggering ?

**Comment by braddialpad** [comment]
Oh I see, you actually have to mouseover it, it doesn't show up automatically when you hit the limit

**Comment by sanjeevdialpad** [comment]
Screenshot 2023-10-26 at 9 47 41 AM
It does show up on over. Do know that it will only be enabled when its negative but in your screenshot it is. just hover over it it should show up.

**Comment by iropolo** [comment]
ohh.. huh, ye it is working I just need to hover it, I expected it will be shown when it is reached the limit

image

**Comment by sanjeevdialpad** [comment]
> Oh I see, you actually have to mouseover it, it doesn't show up automatically when you hit the limit

oh no, it supposed to just show on hover when the user is puzzled why send button is disabled. The disable of send button should be from product side.

File: recipes/conversation_view/message_input/message_input.vue [hunk 1]
```diff
@@ -145,13 +145,24 @@
         <!-- Right content -->
         <div class="d-d-flex">
           <!-- Optionally displayed remaining character counter -->
-          <p
+          <dt-tooltip
             v-if="displayCharacterLimitWarning"
+            :enabled="characterLimitWarningMessage && (characterLimitCount - inputLength < 0)"
+            placement="top-end"
+            :message="characterLimitWarningMessage"
             class="d-fc-error d-mr16 d-as-center dt-message-input--remaining-char"
```
Comment (braddialpad) [inline recipes/conversation_view/message_input/message_input.vue]: Should these classes be on the anchor <p> instead of the tooltip itself?

**Comment by sanjeevdialpad** [comment]
the screenshot is in the ticket showing what the behaviour is: https://dialpad.atlassian.net/browse/DP-81811

**Review by braddialpad — COMMENTED** [review]
Looks pretty good I think, just one question.

**Comment by braddialpad** [comment]
> oh no, it supposed to just show on hover when the user is puzzled why send button is disabled. The disable of send button should be from product side.

Yeah makes sense, I just didn't understand the functionality initally.

**Comment by iropolo** [comment]
I see it okay..., should it have a tooltip also when the warning appear? I mean this case...

image


Something like "You have {x} characters remaining."

**Review by iropolo — APPROVED** [review]
Looks good to me.
Pre-approving it. (Review Brad's comments please)

**Comment by braddialpad** [comment]
> oh no, it supposed to just show on hover when the user is puzzled why send button is disabled. The disable of send button should be from product side.

Not really your issue, more of a design issue, but wouldn't it make sense for the tooltip to show when hovering the send button rather than the red text then? I know if I was a user I'd likely hover the button rather than the red text

File: recipes/conversation_view/message_input/message_input.vue (hunk 1 above)
Comment (sanjeevdialpad) [reply recipes/conversation_view/message_input/message_input.vue]: i moved everything from <p> as i had an issue with it showing up. didnt have to move classes. Moved it back.

**Comment by sanjeevdialpad** [comment]
> > oh no, it supposed to just show on hover when the user is puzzled why send button is disabled. The disable of send button should be from product side.
> 
> Not really your issue, more of a design issue, but wouldn't it make sense for the tooltip to show when hovering the send button rather than the red text then? I know if I was a user I'd likely hover the button rather than the red text

I will pass on the feedback. i felt the same way too. but then you can argue a disabled send button should not be focusable.

**Comment by braddialpad** [comment]
> I will pass on the feedback. i felt the same way too. but then you can argue a disabled send button should not be focusable.

It's actually a very common case and we have guidance for it in our docs
https://dialpad.design/components/button.html#disabled

This definitely doesn't need to hold up this PR but yeah it's a good idea to shoot this question at design I think. It feels a bit off.

**Review by braddialpad — APPROVED** [review]

**Pull request merged**